src/PackageJsonSynchronizer.php

```diff
@@ -76,34 +84,77 @@ private function removePackageJsonLinks(array $packageJson): bool
                     continue;
                 }
 
-                $manipulator = new JsonManipulator(file_get_contents($this->rootDir.'/package.json'));
-                $manipulator->removeSubNode('devDependencies', $name);
-                file_put_contents($this->rootDir.'/package.json', $manipulator->getContents());
-                $didRemoveLink = true;
+                $manipulator->removeSubNode($key, $name);
+                $didChangePackageJson = true;
             }
         }
 
-        return $didRemoveLink;
+        file_put_contents($this->rootDir.'/package.json', $manipulator->getContents());
+
+        return $didChangePackageJson;
     }
 
-    private function addPackageJsonLink(array $phpPackage): bool
+    private function resolvePackageDependencies($phpPackage): array
     {
+        $dependencies = [];
+
         if (!$packageJson = $this->resolvePackageJson($phpPackage)) {
-            return false;
+            return $dependencies;
         }
 
-        $manipulator = new JsonManipulator(file_get_contents($this->rootDir.'/package.json'));
-        $manipulator->addSubNode('devDependencies', '@'.$phpPackage['name'], 'file:'.substr($packageJson->getPath(), 1 + \strlen($this->rootDir), -13));
+        $dependencies['@'.$phpPackage['name']] = 'file:'.substr($packageJson->getPath(), 1 + \strlen($this->rootDir), -13);
+
+        $versionParser = new VersionParser();
```

Comment:
Relying on the Composer version parser to parse npm constraints looks wrong to me. The syntax is not the same between both ecosystems (some parts of the syntax are common, but not everything)

Reply:
and worse than that. Some syntax is the same with different meaning (so the parser would accept it but produce a different result)

Reply:
I agree it's a problem. In the current version it simply generates an invalid package.json the developer has to fix themselves. This hasn't been a common use case for Flex yet so it rarely occurs. What is the intended behavior?

Reply:
Can you explain the problem a bit more that this is solving? Is this if you install 2 packages that both have `chart.js` has a peer dependency... but they one requires v2 and another requires v3?

If so, yea... it's super rare. Well, probably not a reality at all right now. I would vote to simply "do nothing". What I mean is, if you already had `chart.js` in your package.json, leave it with whatever version was there. This is a problem the user will need to resolve and they will see peer dependency warnings from yarn/npm.

Reply:
I wasn't trying to solve it but while writing the PR I noticed it simply generated invalid JSON with 2 conflicting peer dependencies.

Apperantly I forgot the use case for peer dependencies in NPM so awesome suggestion. Thanks :)